Orange/data/domain.py

```diff
@@ -211,6 +211,9 @@ def variables(self):
     def metas(self):
         return self._metas
 
+    def all(self, condition=Variable):
+        return [v for v in chain(self._variables, self._metas) if isinstance(v, condition)]
```

Comment:
`all()` is not good. The idiom means something completely different in Python/NumPy context. `filter()` would be better, `.where()` would be best.

This change should require re-touching _everywhere_ the former (`[]+[]` or `chain`) idiom was used.

Reply:
Ok.

Reply:
I don't like `where`. :)

`all` reads well: I want all discrete variables, hence `domain.all(DiscreteVariable)`. But I agree the name is wrong wrt its usual meaning in Python.

`domain.where(DiscreteVariable)` doesn't read so well and it also means two different things in numpy, one of which is similar to this `Domain.where` (but with different arguments) and one which is completely different.

`domain.filter(DiscreteVariable)` is sort of fine, but it gives an impression that it will return a new instance of `Domain` with just discrete variables.

What about `domain.collect(DiscreteVariable)`? Or `domain.pick(DiscreteVariable)` -- though I prefer `collect`?